app/decorators/stream_item_decorator.rb

```diff
@@ -186,4 +189,26 @@ def new?
   def new_badge
     h.content_tag(:small, h.t('new'), class: 'badge bg-green')
   end
+
+  def avatar_path(person, size=:tn, variation=nil)
+    if person.is_a?(Family)
+      family_avatar_path(person, size)
+    elsif person.is_a?(Group)
+      group_avatar_path(person, size)
+    elsif person.is_a?(Album)
+      album_avatar_path(person, size)
+    else
+      if person.try(:photo).try(:exists?)
+        person.photo.url(size)
+      else
+        size = :large unless size == :tn # we only have only two sizes
+        img = person.try(:gender) == 'Female' ? 'woman' : 'man'
+        if variation == :dark
+          h.image_path("#{img}.dark.#{size}.png")
+        else
+          h.image_path("#{img}.#{size}.jpg")
+        end
+      end
+    end
+  end
```

Comment:
@cessien I think you can just call `avatar_path` without copying this over from the PeopleHelper.... can you try that?